### PR TITLE
Add google tcmalloc to planning server

### PR DESCRIPTION
**Pull request by Levi-Armstrong**
The tcmalloc provide much better memory stability over the standard system version.

### CODE REVIEW

**Comment by Levi-Armstrong** [comment]
This depends on Tesseract [PR](https://github.com/ros-industrial-consortium/tesseract/pull/528).

File: tesseract_process_managers/examples/memory_usage_example.cpp [hunk 1]
```diff
@@ -0,0 +1,145 @@
+
```
Comment (mpowelson) [inline tesseract_process_managers/examples/memory_usage_example.cpp:1]: File doxygen

File: tesseract_process_managers/examples/memory_usage_example.cpp [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+#include <fstream>
+#include <chrono>
+#include <unistd.h>
+#include <string>
+TESSERACT_COMMON_IGNORE_WARNINGS_POP
+
+#include <tesseract_environment/core/environment.h>
+#include <tesseract_environment/ofkt/ofkt_state_solver.h>
+#include <tesseract_command_language/command_language.h>
+#include <freespace_example_program.h>
```
Comment (mpowelson) [inline tesseract_process_managers/examples/memory_usage_example.cpp:13]: Is this correct?

File: tesseract_process_managers/examples/memory_usage_example.cpp [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  return mod_url;
+}
+
+//////////////////////////////////////////////////////////////////////////////
+//
+// process_mem_usage(double &, double &) - takes two doubles by reference,
+// attempts to read the system-dependent data for a process' virtual memory
+// size and resident set size, and return the results in KB.
+//
+// On failure, returns 0.0, 0.0
```
Comment (mpowelson) [inline tesseract_process_managers/examples/memory_usage_example.cpp:54]: Convert to doxygen?

**Review by mpowelson — COMMENTED** [review]
I have not tried this on robot since my code is a branched off of pre-breaking off tesseract_planning.

File: tesseract_process_managers/examples/memory_usage_example.cpp (hunk 2 above)
Comment (Levi-Armstrong) [reply tesseract_process_managers/examples/memory_usage_example.cpp:13]: Copy past content, I will remove unused headers.

**Comment by Levi-Armstrong** [comment]
Also when using tcmalloc it has a set of [tunables](https://gperftools.github.io/gperftools/tcmalloc.html) also. The release rate one is the one I have been using to control memory being release back to the OS.

File: tesseract_process_managers/examples/memory_usage_example.cpp (hunk 2 above)
Comment (Levi-Armstrong) [reply tesseract_process_managers/examples/memory_usage_example.cpp:13]: Oh, I see now what you were talking about. I will change to use quotes so it changes he search behavior.

**Comment by Levi-Armstrong** [comment]
So CI will fail for windows but I will not hold up merging this. I am researching a solution for windows.

**Comment by Levi-Armstrong** [comment]
@mpowelson Any objections to merging this?

**Comment by mpowelson** [comment]
If you've tested it then no

**Comment by Levi-Armstrong** [comment]
Ok, yea we have been using it without any issues for about a week.

**Pull request closed without merging**